x/staking/src/types/domain.rs

```diff
@@ -275,7 +279,7 @@ impl Validator {
             // TODO: infallible in sdk
             Decimal256::from_atomics(amount, 0).unwrap()
         } else {
-            // TODO: check the code, maybe remove unwrap
+            // panics in original implementation
```

Comment:
Don't think that we need to follow cosmos logic on panic in gears. Remember gas case

Reply:
@joneskm What do you think? 
Should we panic if we cannot create/update a coin/share?

I suggest that the panic is correct because it locks node and it cannot update the chain with wrong amount of coins.

Reply:
I have similar case... but it used in `end_block` which can't return errors

Reply:
In this case, I don't think we should panic. This action can legitimately fail e.g. if the amount to delegate cannot fit into a `Decimal256`.